alacritty_terminal/src/selection.rs

```diff
@@ -144,9 +144,19 @@ impl Selection {
                 start == end
                     || (start.side == Side::Left
                         && end.side == Side::Right
-                        && start.point.line == end.point.line
+                        && (start.point.line == end.point.line)
                         && start.point.col == end.point.col + 1)
             },
+            Selection::Block { region: Range { ref start, ref end } } => {
+                // Empty when at most line different or two cells with right+left sides
```

Comment:
I'm not really sure what this comment means. What is "line" referring to here exactly? Also, I think we could be more clear about the right+left logic. Would you mind rewording this a little bit?

Reply:
I think this is the best way to word this shortly without getting into too much detail. If it takes longer to read the comment than the code, there's not much point either.

And I'm not sure why you're asking about what `line` refers to here? Every point has a column and a line, which other interpretation of `line` would make sense here?

Reply:
Yea, every point has a line, but you don't say which point's line you're talking about. In fact I don't see a reference to `line` anywhere below this comment, so it's somewhat unclear what this means to me.

> If it takes longer to read the comment than the code, there's not much point either.

I don't agree with this, and I believe it's the wrong way to think about comments. You should read comments and code together to get a complete understanding. If the code is clear enough on it's own, then there's no reason for a comment, but on the other hand some code requires vastly more comments than it's own syntax.

Reply:
> Yea, every point has a line, but you don't say which point's line you're talking about. In fact I don't see a reference to line anywhere below this comment, so it's somewhat unclear what this means to me.

Because it doesn't matter which line. Because it's a block selection and line number is completely irrelevant for detecting if it's empty. That's exactly what this comment says.

> If the code is clear enough on it's own, then there's no reason for a comment

Fyi, I only added a comment because the other codepath already had one. I don't think there's much that needs to be explained here, so I'd be fine with just removing it.

Reply:
I'd just remove this comment then. But if you want to keep it it that's fine too. It just confused me a bit.

Reply:
No worries, any idea for improvements? Maybe there's something better.